rfcs/0020-project_api_extend_metadata.md

```diff
@@ -0,0 +1,252 @@
+- Feature Name: project_api_extend_metadata
+- Start Date: 2021-09-09
+- RFC PR: [apache/tvm-rfcs#0020](https://github.com/apache/tvm-rfcs/pull/0000)
+- GitHub Issue: [apache/tvm#0020](https://github.com/apache/tvm/issues/0000)
+
+[RFC][Project API] Extend metadata in ProjectOption
+
+# Summary
+[summary]: #summary
+
+Extend metadata associated with project options provided by the Project API.
+
+
+# Motivation
+[motivation]: #motivation
+
+Currently metadata associated with project options provided by the Project API
+is insufficent to allow building easily and automatically command line parsers
+used by CLI tool like TVMC.
+
+The metadata available for the project options, stored in instances of the
+ProjectOption class, A) do not provide a list of the API methods which support
+the options, B) do not allow to determine if the options are required or
+optional, C) and do not provide a default value if one is used by the Project
+API server. As a consequence it complicates the integration with command line
+interfaces that need to create command line arguments based on the project
+options available for a platform.
```

Comment:
Could this be split into a markdown list rather than embedding it into the sentence?
```suggestion
The metadata available for the project options, stored in instances of the ProjectOption class are limited as they do not:
1. Provide a list of the API methods which support the options
1. Allow determination if the options are required or optional
1. Provide a default value if one is used by the Project API server.

As a consequence it complicates the integration with command line interfaces that need to create command line arguments based on the project options available for a platform.
```